CMakeLists.txt

```diff
@@ -133,7 +133,8 @@ IF (NOT CMAKE_SIZEOF_VOID_P)
     MESSAGE(FATAL_ERROR "Kokkos did not configure correctly and failed to validate compiler. The most likely cause is linkage errors during CMake compiler validation. Please consult the CMake error log shown below for the exact error during compiler validation")
   ENDIF()
 ELSEIF (NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
-  MESSAGE(FATAL_ERROR "Kokkos assumes a 64-bit build; i.e., 8-byte pointers, but found ${CMAKE_SIZEOF_VOID_P}-byte pointers instead")
+  MESSAGE(STATUS "Kokkos assumes a 64-bit build, i.e., 8-byte pointers, but found ${CMAKE_SIZEOF_VOID_P}-byte pointers instead; 32-bit builds are experimental and not officially supported.")
+  add_compile_definitions(KOKKOS_32_BIT)
```

Comment:
We normally use `cmakedefine` for more transparency and transitivity. Also please use `KOKKOS_IMPL_32BIT` or similar.

Reply:
I have replaced `KOKKOS_32_BIT` with `KOKKOS_IMPL_32BIT` now. I could not find examples of usage of `cmakedefine` in the source; apparently it is not used atm? But if you tell me what specifically should be done with it, I could do, no issues.

Reply:
`cmake/KokkosCore_config.h.in` defines all macros that are converted from `CMake` variables (using `cmakedefine`). That should also make sure that user installations see `KOKKOS_IMPL_32BIT`.